src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = /[.\-/]/;
+  const newDate = [];
+  const objectDate = {};
+  const splitDate = date.split(oldSeparator);
+  const isPunctuation = (str) => /^[\p{P}\p{S}]+$/u.test(str);
```

Comment:
The `isPunctuation` function is unnecessary because `toFormat` is guaranteed to have the separator as its last element. You can remove this function and directly use `toFormat[toFormat.length - 1]` as the new separator.